ext/date/php_date.c

```diff
@@ -4798,9 +4798,9 @@ PHP_METHOD(DatePeriod, __construct)
 		}
 	}
 
-	if (dpobj->end == NULL && recurrences < 1) {
+	if (dpobj->end == NULL && (recurrences < 1 || ZEND_LONG_INT_OVFL(recurrences))) {
 		zend_string *func = get_active_function_or_method_name();
-		zend_throw_exception_ex(NULL, 0, "%s(): Recurrence count must be greater than 0", ZSTR_VAL(func));
+		zend_throw_exception_ex(NULL, 0, "%s(): Recurrence count must be between 1 and %d", ZSTR_VAL(func), INT_MAX);
```

Comment:
`between 1` is ambiguous between `> 1` or `>= 1`. Please be precise.

Reply:
That's standard wording (and always refers to the closed interval). cc @Girgias

Reply:
We have always used it to mean an inclusive range AFAIK.

Might help to have helper functions for this, similar to what I did for ValueErrors which are empty.

Reply:
should I revert back to the previous wording ?